tests/core/arch-x86/stalker-x86.c

```diff
@@ -3201,3 +3223,174 @@ TESTCASE (ic_var)
+
+  gum_stalker_follow_me (fixture->stalker, fixture->transformer,
+      GUM_EVENT_SINK (fixture->sink));
+  test_try_and_catch (&val);
+  test_check_followed ();
+  gum_stalker_unfollow_me (fixture->stalker);
+
+  if (g_test_verbose())
+    g_print ("val: 0x%08x\n", val);
+
+  test_check_bit (&val, 0);
+  test_check_bit (&val, 2);
+  test_check_bit (&val, 3);
+  test_check_bit (&val, 31);
+  g_assert_cmpuint (val, ==, 0);
+}
+
+TESTCASE (try_and_catch_excluded)
+{
+  guint32 val = 0;
+  const GumMemoryRange range = {
+    .base_address = GPOINTER_TO_SIZE (__cxa_throw),
+    .size = 1
+  };
+
+  gum_stalker_exclude (fixture->stalker, &range);
+
+  fixture->transformer = gum_stalker_transformer_make_from_callback (
+      callback_at_end, &val, NULL);
+
+  gum_stalker_follow_me (fixture->stalker, fixture->transformer,
+      GUM_EVENT_SINK (fixture->sink));
+  test_try_and_catch (&val);
+  test_check_followed ();
+  gum_stalker_unfollow_me (fixture->stalker);
+
+  if (g_test_verbose())
+    g_print ("val: 0x%08x\n", val);
+
+  test_check_bit (&val, 0);
+  test_check_bit (&val, 2);
+  test_check_bit (&val, 3);
+  test_check_bit (&val, 31);
+  g_assert_cmpuint (val, ==, 0);
+}
+
+TESTCASE (try_and_dont_catch)
+{
+  guint32 val = 0;
+
+  fixture->transformer = gum_stalker_transformer_make_from_callback (
+      callback_at_end, &val, NULL);
+
+  gum_stalker_follow_me (fixture->stalker, fixture->transformer,
+      GUM_EVENT_SINK (fixture->sink));
+  test_try_and_dont_catch (&val);
+  test_check_followed ();
+  gum_stalker_unfollow_me (fixture->stalker);
+
+  if (g_test_verbose())
+    g_print ("val: 0x%08x\n", val);
+
+  test_check_bit (&val, 0);
+  test_check_bit (&val, 1);
+  test_check_bit (&val, 2);
+  test_check_bit (&val, 5);
+  test_check_bit (&val, 6);
+  test_check_bit (&val, 7);
+  test_check_bit (&val, 11);
+  test_check_bit (&val, 31);
+  g_assert_cmpuint (val, ==, 0);
+
+}
+
+TESTCASE (try_and_dont_catch_excluded)
+{
+  guint32 val = 0;
+  const GumMemoryRange range = {
+    .base_address = GPOINTER_TO_SIZE (__cxa_throw),
+    .size = 1
+  };
+
+  gum_stalker_exclude (fixture->stalker, &range);
+
+  fixture->transformer = gum_stalker_transformer_make_from_callback (
+      callback_at_end, &val, NULL);
+
+  gum_stalker_follow_me (fixture->stalker, fixture->transformer,
+      GUM_EVENT_SINK (fixture->sink));
+  test_try_and_dont_catch (&val);
+  test_check_followed ();
+  gum_stalker_unfollow_me (fixture->stalker);
+
+  if (g_test_verbose())
+    g_print ("val: 0x%08x\n", val);
+
+  test_check_bit (&val, 0);
+  test_check_bit (&val, 1);
+  test_check_bit (&val, 2);
+  test_check_bit (&val, 5);
+  test_check_bit (&val, 6);
+  test_check_bit (&val, 7);
+  test_check_bit (&val, 11);
+  test_check_bit (&val, 31);
+  g_assert_cmpuint (val, ==, 0);
+}
+
+static void
+callback_at_end (GumStalkerIterator * iterator,
+                GumStalkerOutput * output,
+                gpointer user_data)
+{
+  guint32 * val = (guint32 *) user_data;
```

Comment:
Redundant cast. (Since this is C, not C++, `void *` to `any *` does not require a cast.)